src/App.tsx

```diff
@@ -1,27 +1,139 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
+import {
+  Container,
+  Stack,
+  Button,
+  Typography,
+} from '@mui/material';
+
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
+
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+
+  enum ButtonVariant {
+    outlined = 'outlined',
+    contained = 'contained',
+  }
+
+  const [buttonVariants, setButtonVariants] = useState({
+    all: ButtonVariant.outlined,
+    firstFive: ButtonVariant.outlined,
+    red: ButtonVariant.outlined,
+  });
+
+  const handleLoadAll = async () => {
+    if (buttonVariants.all !== ButtonVariant.outlined) {
+      setGoods([]);
+      setButtonVariants({
+        all: ButtonVariant.outlined,
+        firstFive: ButtonVariant.outlined,
+        red: ButtonVariant.outlined,
+      });
+
+      return;
+    }
+
+    setGoods(await getAll());
+    setButtonVariants({
+      all: ButtonVariant.contained,
+      firstFive: ButtonVariant.outlined,
+      red: ButtonVariant.outlined,
+    });
```

Comment:
```suggestion
    setButtonVariants((prevState) => ({
      ...prevState,
      all: ButtonVariant.contained,
    }));
```

Reply:
@dffUqp I didn't change solution in that situation, if I use your advice I can press all 3 buttons at the same time (of course in list I see last result, but all 3 buttons can be in `contained` state in at the same time) 

![image](https://github.com/mate-academy/react_dynamic-list-of-goods/assets/105503953/82628ac7-2bf9-42f6-891d-4e7bae988226)

Reply:
oh, sorry😢

try this 
```
    setButtonVariants({
       ...initialButtonVariants,
      all: ButtonVariant.contained
    });
```

Reply:
@dffUqp WOW, it looks awesome, honestly I didn't think of that at all. Thank you very much